scripts/graphdb_setup.sh

```diff
@@ -0,0 +1,176 @@
+handle_passed_args_count
+assign_positional_args 1 "${_positionals[@]}"
+
+# OTHER STUFF GENERATED BY Argbash
+
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+# [ <-- needed because of Argbash
+
+set -euo pipefail
+
+# Reassign command line args to more readable named variables
+ENV_FILE_PATH=$_arg_env_file_path
+ADMIN_PASS=$_arg_admin_pass
+RUN_USER_SETUP=$_arg_run_user_setup
+
+# Set the environment variables in the shell, to use in the script
+source "${ENV_FILE_PATH}"
+echo "Environment variables have been set from ${ENV_FILE_PATH}."
+
+# Extract just the database name
+DB_NAME="${NB_GRAPH_DB#repositories/}"
+NB_GRAPH_PORT_HOST=${NB_GRAPH_PORT_HOST:-7200}
+
+# Get the directory of this script to be able to find the data-config_template.ttl file
+SCRIPT_DIR=$(dirname "$0")
+
+echo "The GraphDB server is being accessed at http://localhost:${NB_GRAPH_PORT_HOST}."
+
+##### First time GraphDB setup #####
+
+if [ "${RUN_USER_SETUP}" = "on" ]; then
+    # 1. Change database admin password and allow only authenticated users access
+    echo "First time GraphDB user setup enabled."
+    echo "Changing the admin password..."
+    curl -X PATCH --header 'Content-Type: application/json' http://localhost:${NB_GRAPH_PORT_HOST}/rest/security/users/admin -d "{\"password\": \""${ADMIN_PASS}"\"}"
+    echo "Allowing only authenticated users access..."
+    curl -X POST --header 'Content-Type: application/json' -d true http://localhost:${NB_GRAPH_PORT_HOST}/rest/security
+
+    # 2. Create a new database user
+    echo "Creating a new database user ${NB_GRAPH_USERNAME}..."
+    curl -X POST --header 'Content-Type: application/json' -u "admin:${ADMIN_PASS}" -d @- http://localhost:${NB_GRAPH_PORT_HOST}/rest/security/users/${NB_GRAPH_USERNAME} <<EOF
+    {
+        "username": "${NB_GRAPH_USERNAME}",
+        "password": "${NB_GRAPH_PASSWORD}"
+    } 
+EOF
+fi
+
+
+##### Database setup #####
+
+# 3. Create and save custom configuration file for the new database
+# TODO: Should we add a suffix to data-config.ttl for the db name?
+echo "Creating a GraphDB configuration file (./data-config.ttl) for the new database ${DB_NAME}: ..."
+sed 's/rep:repositoryID "my_db" ;/rep:repositoryID "'"${DB_NAME}"'" ;/' ${SCRIPT_DIR}/data-config_template.ttl > data-config.ttl
+
+# 4. Create a new database and give newly created user access permission
+# Assumes data-config.ttl is in the same directory as this script!
+echo "Creating the GraphDB database ${DB_NAME}..."
+curl -X PUT -u "admin:${ADMIN_PASS}" http://localhost:${NB_GRAPH_PORT_HOST}/${NB_GRAPH_DB} --data-binary "@data-config.ttl" -H "Content-Type: application/x-turtle"
+echo "Granting user ${NB_GRAPH_USERNAME} read/write permissions to database ${DB_NAME}..."
+curl -X PUT --header 'Content-Type: application/json' -d "
+{\"grantedAuthorities\": [\"WRITE_REPO_${DB_NAME}\",\"READ_REPO_${DB_NAME}\"]}" http://localhost:${NB_GRAPH_PORT_HOST}/rest/security/users/${NB_GRAPH_USERNAME} -u "admin:${ADMIN_PASS}"
```

Comment:
so this is the thing I killed the openNeuro API with yesterday. What will happen here if I already have a database in the graph instance (i.e. I am only running this to add 1 new database)? 

Suppose
- `{DB_EXISTS_ALREADY}` is my existing database AND 
- `{NB_GRAPH_USERNAME}` has access to it AND
- `{DB_NAME}` is the new one I want to add
- Then I think this command will remove access to `{DB_EXISTS_ALREADY}`  for `{NB_GRAPH_USERNAME}` 

Could you check if that's true - and fix if yes

Reply:
Decision was: add warning with re-prompt for confirmation when user selects to just "make a new database". Purpose of warning is to remind user that this may remove existing permissions